pkg/controllers/disruption/controller.go

```diff
@@ -126,8 +109,10 @@ func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 
 	// Karpenter taints nodes with a karpenter.sh/disruption taint as part of the disruption process
 	// while it progresses in memory. If Karpenter restarts during a disruption action, some nodes can be left tainted.
-	// Idempotently remove this taint from candidates before continuing.
-	if err := c.requireNoScheduleTaint(ctx, false, c.cluster.Nodes()...); err != nil {
+	// Idempotently remove this taint from candidates that are not in the orchestration queue before continuing.
+	if err := state.RequireNoScheduleTaint(ctx, c.kubeClient, false, lo.Filter(c.cluster.Nodes(), func(s *state.StateNode, _ int) bool {
+		return c.Queue.CanAdd(s.ProviderID()) == nil
```

Comment:
```suggestion
		return c.Queue.CanAdd(s.ProviderID()) == nil
```

Does it make sense for this to be `CanAdd()` now or should it just be `Has()` to say that this element is in the queue?

Reply:
I could go either way. Happy to change to Has() if you think that's better, but we also call this in queue.Add() function, so I think it makes sense there more as CanAdd()

Reply:
I'm fine to keep it as-is. I'll let you make the call.